tools/gen_fresh_dirinfo.py

```diff
@@ -715,6 +796,49 @@ def compute_churn(
     return churn
 
 
+
+def generate_certificate(namespace: Namespace) -> None:
+    """
+    Generate a certificate for a custom authority.
+
+    :param namespace: namespace containing parsed arguments.
+    """
+    password = environ.get(DIR_AUTH_PASSWORD_ENV, None)
+    if password is None:
+        raise Exception(f"No password provided as environment variable {DIR_AUTH_PASSWORD_ENV}.")
+
+    authority_identity_key_path: Path = namespace.authority_identity_key
+    authority_signing_key_path: Path = namespace.authority_signing_key
+    authority_certificate_path: Path = namespace.authority_certificate
+    authority_v3ident_path: Path = namespace.authority_v3ident
+    authority_name: Path = namespace.authority_name
+    certificate_lifetime: int = namespace.certificate_lifetime
+
+    if authority_identity_key_path.exists():
+        create_new_identity = False
+        reuse_signing_key = authority_signing_key_path.exists()
+    else:
+        create_new_identity = True
+        reuse_signing_key = False
```

Comment:
boolean logic is fun :)

```suggestion
    create_new_identity = !authority_identity_key_path.exists()
    reuse_signing_key = !create_new_identity and authority_signing_key_path.exists()
```

Reply:
Good point!